test/spec/SearchApi/Providers/SolrSearchSpec.php

```diff
@@ -32,10 +32,13 @@ function it_should_return_null_when_keywords_empty() {
     $result->shouldBeNull();
   }
 
-  function it_should_return_search_result_when_keywords_not_empty() {
-    $keywords = array( new Models\SearchTerm( 'test' ) );
-    $result = $this->query( $keywords );
+  function it_should_return_search_result_when_keywords_not_empty( QueryBuilder $queryBuilder, HttpClient $httpClient ) {
+    $this->beConstructedWith( $queryBuilder, $httpClient, 'url/' );
+    $queryBuilder->build( 'test', null )->willReturn( 'test_query' );
+    $httpClient->get( 'url/test_query' )->shouldBeCalled()->willReturn( '{"response":{"numFound":1,"start":0,"docs":[{"id":"testid","author":"testauthor","date":"testdate","content":"testcontent"}]}}' );
```

Comment:
Move `'{"response":{"numFound":1,"start":0,"docs":[{"id":"testid","author":"testauthor","date":"testdate","content":"testcontent"}]}}'` to its own variable just for readability